mpGraphql Exception whitelist not working. NullPointerException is thrown by mpConfig

**Describe the bug**
Hello, I am using microprofile-graphql on openliberty and everything goes well except for the exception whitelisting mechanism via microprofile config described [here](https://download.eclipse.org/microprofile/microprofile-graphql-1.0.2/microprofile-graphql.html#_unchecked_exceptions).

Once enabled:

openliberty stdout log

```
[INFO] [ERROR   ] null
[INFO] CWMCG0007E: A Converter of type interface java.util.List threw an exception. The input string was: "com.slabs.gsaas.gw.infra.clients.InternalServiceException". The exception is: null.
```
messages.log

```
[7/14/20, 8:35:56:748 CEST] 00000082 io.smallrye.graphql.servlet.ExecutionServlet                 E null
com.ibm.ws.microprofile.config.interfaces.ConfigException: CWMCG0007E: A Converter of type interface java.util.List threw an exception. The input string was: "com.slabs.gsaas.gw.infra.clients.InternalServiceException". The exception is: null.
	at com.ibm.ws.microprofile.config.impl.ConversionManager.simpleConversion(ConversionManager.java:78)
	at com.ibm.ws.microprofile.config.impl.ConversionManager.convert(ConversionManager.java:140)
	at com.ibm.ws.microprofile.config.impl.ConversionManager.convert(ConversionManager.java:103)
	at com.ibm.ws.microprofile.config14.impl.Config14Impl.getCachedSourcedValue(Config14Impl.java:79)
	at com.ibm.ws.microprofile.config14.impl.Config14Impl.getSourcedValue(Config14Impl.java:62)
	at com.ibm.ws.microprofile.config.impl.AbstractConfig.getOptionalValue(AbstractConfig.java:67)
	at com.ibm.ws.io.smallrye.graphql.component.ConfigFacade.getOptionalValue(ConfigFacade.java:37)
	at com.ibm.ws.io.smallrye.graphql.component.GraphQLServletContainerInitializer$1.getWhiteList(GraphQLServletContainerInitializer.java:139)
	at io.smallrye.graphql.execution.error.ExceptionHandler.<init>(ExceptionHandler.java:26)
	at io.smallrye.graphql.execution.ExecutionService.getGraphQL(ExecutionService.java:139)
	at io.smallrye.graphql.execution.ExecutionService.execute(ExecutionService.java:68)
	at io.smallrye.graphql.servlet.ExecutionServlet.handleInput(ExecutionServlet.java:86)
	at io.smallrye.graphql.servlet.ExecutionServlet.doPost(ExecutionServlet.java:71)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:706)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:791)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.service(ServletWrapper.java:1230)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:729)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:426)
	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1226)
	at com.ibm.ws.webcontainer.webapp.WebApp.handleRequest(WebApp.java:5037)
	at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.handleRequest(DynamicVirtualHost.java:314)
	at com.ibm.ws.webcontainer.WebContainer.handleRequest(WebContainer.java:1007)
	at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.run(DynamicVirtualHost.java:279)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink$TaskWrapper.run(HttpDispatcherLink.java:1134)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.wrapHandlerAndExecute(HttpDispatcherLink.java:415)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.ready(HttpDispatcherLink.java:374)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleDiscrimination(HttpInboundLink.java:546)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleNewRequest(HttpInboundLink.java:480)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.processRequest(HttpInboundLink.java:345)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.ready(HttpInboundLink.java:316)
	at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.sendToDiscriminators(NewConnectionInitialReadCallback.java:167)
	at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.complete(NewConnectionInitialReadCallback.java:75)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.requestComplete(WorkQueueManager.java:504)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.attemptIO(WorkQueueManager.java:574)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.workerRun(WorkQueueManager.java:958)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager$Worker.run(WorkQueueManager.java:1047)
	at com.ibm.ws.threading.internal.ExecutorServiceImpl$RunnableWrapper.run(ExecutorServiceImpl.java:239)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.NullPointerException
	at java.base/java.lang.reflect.Array.newArray(Native Method)
	at java.base/java.lang.reflect.Array.newInstance(Array.java:78)
	at com.ibm.ws.microprofile.config.impl.ConversionManager.convertArray(ConversionManager.java:279)
	at com.ibm.ws.microprofile.config.impl.ConversionManager.convertArray(ConversionManager.java:272)
	at com.ibm.ws.microprofile.config12.converters.ListConverter.convert(ListConverter.java:40)
	at com.ibm.ws.microprofile.config12.converters.ListConverter.convert(ListConverter.java:23)
	at com.ibm.ws.microprofile.config.impl.ConversionManager.simpleConversion(ConversionManager.java:62)
	... 39 more
```

**Steps to Reproduce**
Add the following option to microprofile-config:
```
mp.graphql.exceptionsWhiteList=com.mycompany.exceptions.InternalServiceException
```

**Expected behavior**
The application runs correctly and when an `InternalServiceException` occurs, the exception message overrides the default error message returned to the client

**Diagnostic information:**
 - OpenLiberty Version: 20.0.0.6
 - Java Version: 11
- server.xml

```
<server description="Gateway">

    <featureManager>
        <feature>mpRestClient-1.4</feature>
        <feature>mpGraphQL-1.0</feature>
        <feature>jaxrs-2.1</feature>
        <feature>jsonb-1.0</feature>
        <feature>mpConfig-1.4</feature>
        <feature>cdi-2.0</feature>
        <feature>servlet-4.0</feature>
    </featureManager>

    <httpEndpoint host="*" httpPort="${http.port}" httpsPort="${https.port}"
                  id="defaultHttpEndpoint"/>

    <webApplication location="gw.war" contextRoot="/gw"/>
</server>
```

**Additional context**
Issue raised also [here](https://github.com/eclipse/microprofile-graphql/issues/280#issue-656375812).

--- a/dev/com.ibm.ws.microprofile.config.1.2/src/com/ibm/ws/microprofile/config12/converters/ListConverter.java
+++ b/dev/com.ibm.ws.microprofile.config.1.2/src/com/ibm/ws/microprofile/config12/converters/ListConverter.java
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2018 IBM Corporation and others.
+ * Copyright (c) 2018, 2020 IBM Corporation and others.
  * All rights reserved. This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License v1.0
  * which accompanies this distribution, and is available at
@@ -36,9 +36,16 @@ public class ListConverter extends BuiltInConverter implements ExtendedGenericCo
 
     /** {@inheritDoc} */
     @Override
-    public <T> List<T> convert(String rawString, Class<T> genericType, ConversionManager conversionManager, ClassLoader classLoader) {
-        T[] array = conversionManager.convertArray(rawString, genericType);
-        List<T> list = Arrays.asList(array);
+    public <T> List<Object> convert(String rawString, Class<T> genericTypeNullable, ConversionManager conversionManager, ClassLoader classLoader) {
+        // If the generic type is not specified, use String by default.
+        // E.g. if the user called: Config.getConfig().getValue("foo", List.class);
+        Class<?> genericType = genericTypeNullable;
+        if (genericType == null) {
+            genericType = String.class;
+        }
+        Object[] array = conversionManager.convertArray(rawString, genericType);
+        List<Object> list = Arrays.asList(array);
+
         return list;
     }
 }

--- a/dev/com.ibm.ws.microprofile.config.1.2/src/com/ibm/ws/microprofile/config12/converters/SetConverter.java
+++ b/dev/com.ibm.ws.microprofile.config.1.2/src/com/ibm/ws/microprofile/config12/converters/SetConverter.java
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2018 IBM Corporation and others.
+ * Copyright (c) 2018, 2020 IBM Corporation and others.
  * All rights reserved. This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License v1.0
  * which accompanies this distribution, and is available at
@@ -39,9 +39,16 @@ public class SetConverter extends BuiltInConverter implements ExtendedGenericCon
 
     /** {@inheritDoc} */
     @Override
-    public <T> Set<T> convert(String rawString, Class<T> genericType, ConversionManager conversionManager, ClassLoader classLoader) {
-        T[] array = conversionManager.convertArray(rawString, genericType);
-        Set<T> set = new HashSet<T>(Arrays.asList(array));
+    public <T> Set<Object> convert(String rawString, Class<T> genericTypeNullable, ConversionManager conversionManager, ClassLoader classLoader) {
+        // If the generic type is not specified, use String by default.
+        // E.g. if the user called: Config.getConfig().getValue("foo", Set.class);
+        Class<?> genericType = genericTypeNullable;
+        if (genericType == null) {
+            genericType = String.class;
+        }
+        Object[] array = conversionManager.convertArray(rawString, genericType);
+        Set<Object> set = new HashSet<>(Arrays.asList(array));
+
         return set;
     }
 }

--- a/dev/com.ibm.ws.microprofile.config.1.2/test/src/com/ibm/ws/microprofile/config12/test/converters/ConversionUtil.java
+++ b/dev/com.ibm.ws.microprofile.config.1.2/test/src/com/ibm/ws/microprofile/config12/test/converters/ConversionUtil.java
@@ -1,0 +1,39 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.microprofile.config12.test.converters;
+
+import static org.junit.Assert.assertEquals;
+
+import java.lang.reflect.Type;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.spi.ConfigProviderResolver;
+
+import com.ibm.ws.microprofile.config.interfaces.WebSphereConfig;
+
+public class ConversionUtil extends AbstractConfigTest {
+
+    public static <T> void assertConversion(String key, String rawString, T expected, Class<T> type) {
+        System.setProperty(key, rawString);
+        Config config = ConfigProviderResolver.instance().getBuilder().addDefaultSources().build();
+
+        T value = config.getValue(key, type);
+        assertEquals(expected, value);
+    }
+
+    public static void assertTypedConversion(String key, String rawString, Object expected, Type type) {
+        System.setProperty(key, rawString);
+        WebSphereConfig config = (WebSphereConfig) ConfigProviderResolver.instance().getBuilder().addDefaultSources().build();
+
+        Object value = config.getValue(key, type);
+        assertEquals(expected, value);
+    }
+}

--- a/dev/com.ibm.ws.microprofile.config.1.2/test/src/com/ibm/ws/microprofile/config12/test/converters/ListConversionTest.java
+++ b/dev/com.ibm.ws.microprofile.config.1.2/test/src/com/ibm/ws/microprofile/config12/test/converters/ListConversionTest.java
@@ -1,0 +1,68 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.microprofile.config12.test.converters;
+
+import java.lang.reflect.Type;
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.Test;
+
+public class ListConversionTest extends AbstractConfigTest {
+
+    @SuppressWarnings("unused")
+    private List<Integer> listIntField;
+
+    @Test
+    public void testList() {
+        String one = "One";
+        String two = "Two";
+        String three = "Three";
+        String four = "Four";
+
+        String input = one + "," + two + "," + three + "," + four;
+        List<String> expected = Arrays.asList(one, two, three, four);
+
+        ConversionUtil.assertConversion("testList", input, expected, List.class);
+    }
+
+    @Test
+    public void testSingleElementList() {
+        String input = "One";
+        List<String> expected = Arrays.asList("One");
+
+        ConversionUtil.assertConversion("testSingleElementList", input, expected, List.class);
+    }
+
+    @Test
+    public void testEmptyList() {
+        String input = "";
+        List<String> expected = Arrays.asList("");
+
+        ConversionUtil.assertConversion("testEmptyList", input, expected, List.class);
+    }
+
+    @Test
+    public void testNonStringList() throws NoSuchFieldException, SecurityException {
+        int one = 1;
+        int two = 2;
+        int three = 3;
+        int four = 4;
+
+        String input = one + "," + two + "," + three + "," + four;
+        List<Integer> expected = Arrays.asList(1, 2, 3, 4);
+
+        Type listIntType = ListConversionTest.class.getDeclaredField("listIntField").getGenericType();
+
+        ConversionUtil.assertTypedConversion("testNonStringList", input, expected, listIntType);
+    }
+
+}

--- a/dev/com.ibm.ws.microprofile.config.1.2/test/src/com/ibm/ws/microprofile/config12/test/converters/SetConversionTest.java
+++ b/dev/com.ibm.ws.microprofile.config.1.2/test/src/com/ibm/ws/microprofile/config12/test/converters/SetConversionTest.java
@@ -1,0 +1,69 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.microprofile.config12.test.converters;
+
+import java.lang.reflect.Type;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.junit.Test;
+
+public class SetConversionTest extends AbstractConfigTest {
+
+    @SuppressWarnings("unused")
+    private Set<Integer> setIntField;
+
+    @Test
+    public void testSet() {
+        String one = "One";
+        String two = "Two";
+        String three = "Three";
+        String four = "Four";
+
+        String input = one + "," + two + "," + three + "," + four;
+        Set<String> expected = new HashSet<>(Arrays.asList(one, two, three, four));
+
+        ConversionUtil.assertConversion("testSet", input, expected, Set.class);
+    }
+
+    @Test
+    public void testSingleElementSet() {
+        String input = "One";
+        Set<String> expected = new HashSet<>(Arrays.asList("One"));
+
+        ConversionUtil.assertConversion("testSingleElementSet", input, expected, Set.class);
+    }
+
+    @Test
+    public void testEmptySet() {
+        String input = "";
+        Set<String> expected = new HashSet<>(Arrays.asList(""));
+
+        ConversionUtil.assertConversion("testEmptySet", input, expected, Set.class);
+    }
+
+    @Test
+    public void testNonStringSet() throws NoSuchFieldException, SecurityException {
+        int one = 1;
+        int two = 2;
+        int three = 3;
+        int four = 4;
+
+        String input = one + "," + two + "," + three + "," + four;
+        Set<Integer> expected = new HashSet<>(Arrays.asList(1, 2, 3, 4));
+
+        Type setIntType = SetConversionTest.class.getDeclaredField("setIntField").getGenericType();
+
+        ConversionUtil.assertTypedConversion("testNonStringSet", input, expected, setIntType);
+    }
+
+}

